tutorials/en/vectara-beginner-app-tutorial-showcase-the-creation-of-vectara-app-in-legal-use-case.mdx

```diff
@@ -0,0 +1,380 @@
+- **Billing Management**: Keep a tab on your account usage and manage billing details ensuring uninterrupted service as you sail through the Vectara ecosystem.
+
+
+In this section, we've skimmed the surface of Vectara's offerings. As we delve deeper into our chosen use case in the next section, the utility and power of Vectara will unfold further, painting a clearer picture of how it can be harnessed for Customer Support applications.
+
+---
+
+
+# Introduction to Legal Consultation Application 🏛️
+
+Embark on an insightful journey through this tutorial, where we unveil the Legal Consultation Application meticulously crafted using Streamlit, Vectara, and Langchain. This innovative application is engineered to demystify the legal consultation process for individuals or entities in need of legal guidance. With just a simple PDF document upload, users are ushered into a realm of instantaneous, automated consultations based on the document's contents.
+
+## Legal Consultation 📜
+
+The spotlight of this tutorial is a use case deeply rooted in the Legal domain. Amidst a burgeoning demand for swift and accessible legal consultations, this application emerges as a beacon of convenience. By harnessing the magic of automation and artificial intelligence, it offers preliminary legal advice derived from the uploaded documents, making legal assistance merely a click away.
+
+## Concept and Structure of the Application 🛠️
+
+The essence of this application is to provide a user-centric platform where obtaining legal consultations is a breeze with just a PDF document upload. The collaborative prowess of Streamlit, Vectara, and Langchain serves as the bedrock of this application, orchestrating a robust and intuitive environment.
+
+### Streamlit
+Streamlit, an open-source app framework, is the mastermind behind the interactive web interface of our application. It fosters a seamless user journey with widgets for file upload and elegantly unfurls the consultation output to the user, making the interface a joy to navigate.
+
+### Vectara 🧠
+Vectara is the linchpin that elevates the application's capabilities to a new zenith. As a semantic search software company, Vectara is on a mission to redefine search by leveraging artificial intelligence and neural network technologies for natural language processing. It facilitates a deeper understanding of user queries and provides extraordinarily relevant responses. In the scope of our application, Vectara processes the legal documents uploaded by users, delving into the semantics to extract crucial legal insights that form the basis of the automated consultations provided. The integration of Vectara transforms the application into a powerhouse of semantic search, ensuring users receive precise and relevant legal advice based on their documents.
+
+### Langchain 🤖
```

Comment:
Using LangChain is fine and should work. But honestly for this simple application it's not necessary - why not call the Vectara query API directly? This will remove a dependency on LangChain and make the application easier. Thoughts?

Reply:
I've given this matter considerable thought. While the advanced version does offer greater modularity and customization options, I recognize the potential benefits of catering to beginners, particularly those interested in LangChain given its rising popularity.

There are participants who have submitted projects based on LangChain in various hackathons. For them, and for any beginner looking to create something using LangChain, this approach could simplify the process of taking their project to the next level. People who are new to REST APIs may find it challenging to format the output correctly, and that's a key consideration in this context.

Additionally, I am contemplating linking this beginner-friendly version to the advanced one, providing users with an accessible pathway to scale their projects as they grow more confident and capable.

Why langchain though , according to me:

Ease of Creating Chatbots: With LangChain, creating chatbots will be very easy, especially when utilizing Streamlit callbacks handler for streaming. This solves the entire use case for chatbot applications.
Quick Development of Agents: Agents can be developed swiftly,and tools can be created while using function calling , to like create something even complex.
Simplified Introduction to Vectara: Beginners won’t have to delve into the complexities of indexing and searching APIs immediately. However, they will still gain a glimpse into Vectara’s capabilities and understand the basics of what's happening behind the scenes.

I am open to suggestions , but I think linking Beginner and Advanced in such a way that both kinds of participants can utilize this fully and can switch easily.

Reply:
Got it - yes this makes sense. LangChain is fine let's leave it in. Any way we can perhaps demonstrate here both options: LangChain and LlamaIndex? They are somewhat competing with each other and both provide a "beginner" option.

Reply:
We cannot add llama index because it is just using Vectara as a Vector store , nothing more than that.
Also the things llama index does to make it efficient is already done in vectara itself